### PR TITLE
feat: Allow scaling using redis stream length (#4277)

### PR DESCRIPTION
**Pull request by jkbmdk**
Make the redis stream scaler optionally dependent on the XLEN parameter

### Checklist

- [x] When introducing a new scaler, I agree with the [scaling governance policy](https://github.com/kedacore/governance/blob/main/SCALERS.md)
- [x] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [x] Tests have been added
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)
- [x] A PR is opened to update our Helm chart ([repo](https://github.com/kedacore/charts)) *(if applicable, ie. when deployment manifests are modified)*
- [x] A PR is opened to update the documentation on ([repo](https://github.com/kedacore/keda-docs)) *(if applicable)*
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Relates to  #4277
Relates to  https://github.com/kedacore/test-tools/pull/117
Relates to  https://github.com/kedacore/keda-docs/pull/1094

### CODE REVIEW

File: pkg/scalers/redis_streams_scaler.go [hunk 1]
```diff
@@ -185,28 +209,34 @@ func parseRedisStreamsMetadata(config *ScalerConfig, parseFn redisAddressParser)
 		meta.connectionInfo.unsafeSsl = parsedVal
 	}
 
-	meta.targetPendingEntriesCount = defaultTargetPendingEntriesCount
```
Comment (JorTurFer) [inline pkg/scalers/redis_streams_scaler.go:188]: We should maintain the default value as it's documented and it has been there for multiple versions. To remove it we have to follow the deprecation policy

**Review by JorTurFer — COMMENTED** [review]
The code looks good! Thanks for the improvement ❤️ 
I'll review the e2e tests once the PR in test-tools is merged 😄

File: pkg/scalers/redis_streams_scaler.go (hunk 1 above)
Comment (jkbmdk) [reply pkg/scalers/redis_streams_scaler.go:188]: Default value does not work in main branch, if metadata has not `pendingEntriesCount` key, the `ErrRedisMissingPendingEntriesCount` error is rised. I'll bring it back and make it work

**Comment by jkbmdk** [comment]
Hey @JorTurFer , is there anything else I can do to help introduce that feature?

**Comment by JorTurFer** [comment]
> Hey @JorTurFer , is there anything else I can do to help introduce that feature?

I think that only the conflict is changelog is left here

**Pull request merged**